src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx

```diff
@@ -0,0 +1,201 @@
+      </View>
+    );
+  }
+
+  const clearValues = () => {
+    setCanisterId('');
+    setStandard('DIP20');
+    setLoading(false);
+    setTokenError(undefined);
+    setCanisterIdError(false);
+  };
+
+  const handleClose = () => {
+    modalRef?.current?.close();
+    clearValues();
+  };
+
+  const handleIdChange = (text: string) => {
+    setCanisterId(text);
+    setCanisterIdError(!validateCanisterId(text));
+    setTokenError(undefined);
+  };
+
+  const handleStandardChange = useCallback((selected: FungibleStandard) => {
+    setStandard(selected);
+    setTokenError(undefined);
+  }, []);
+
+  const handleStandardPress = () => {
+    Keyboard.dismiss();
+    optionsRef?.current?.open();
+  };
+
+  const handleLinkPress = () => {
+    Linking.canOpenURL(customTokensUrl).then(() =>
+      Linking.openURL(customTokensUrl)
+    );
+  };
+
+  const handleSubmit = () => {
+    Keyboard.dismiss();
+    if (validateCanisterId(canisterId)) {
+      setLoading(true);
+      setCanisterIdError(false);
+      dispatch(
+        getTokenInfo({
+          token: { canisterId, standard },
+          onSuccess: res => {
+            onSelectedToken(res.token);
+            setLoading(false);
+          },
+          onError: (err: string) => {
+            setTokenError(getError(err));
+            setLoading(false);
+          },
+        })
+      );
+    } else {
+      setCanisterIdError(true);
+    }
+  };
+
+  const standardList = useMemo(
+    () => [
+      {
+        id: 1,
+        label: 'DIP20',
+        onPress: () => handleStandardChange('DIP20'),
+      },
+      {
+        id: 2,
+        label: 'EXT',
+        onPress: () => handleStandardChange('EXT'),
+      },
+    ],
+    [handleStandardChange]
+  );
+
+  return (
+    <Modal modalRef={modalRef} adjustToContentHeight onClosed={clearValues}>
+      <Header
+        center={
+          <Text style={styles.title} type="subtitle3">
+            {t('addToken.customTokenTitle')}
+          </Text>
+        }
+        right={<ActionButton label={t('common.close')} onPress={handleClose} />}
+      />
+      <View style={styles.container}>
+        <TextInput
+          placeholder={t('addToken.customTokenId')}
+          value={canisterId}
+          onChangeText={handleIdChange}
+          blurOnSubmit
+          onBlur={Keyboard.dismiss}
+          error={canisterIdError || showCanisterError}
+        />
+        {showCanisterError &&
+          renderError(tokenError.message, tokenError.showMore)}
+        <Button
+          text={standard || t('addToken.customTokenStandard')}
+          onPress={handleStandardPress}
+          buttonStyle={[
+            styles.standardButton,
+            showStandardError && styles.standardButtonError,
+          ]}
+          textStyle={
+            standard ? styles.standardText : styles.standardTextPlaceholder
+          }
+          iconStyle={styles.standardIcon}
+          iconName="chevronRight"
+          iconProps={{ height: 18 }}
```

Comment:
extract into styles

Reply:
the thing here is that they are not styles, they need to be sent as an object. 
In the future i would like to remove that icon logic and directly send the icon as prop like this:

`left={<Icon ... />}